.github/setup_python/action.yml

```diff
@@ -1,27 +1,56 @@
 name: "Setup Python"
 description: "Sets up Python for use in actions with caching and copy localsettings"
 
+inputs:
+  install-dev-dependencies:
+    description: "if disabled this will only install runtime dependencies"
+    required: false
+    default: true
+
 runs:
   using: "composite"
   steps:
+    - name: Compute requirements files
+      run: |
+        cache_files="requirements.txt"
+        requirements_file="requirements.txt"
+        if [[ ${{ inputs.install-dev-dependencies }} == 'true' ]]; then
+          cache_files="requirements*.txt"
+          requirements_file="requirements-dev.txt"
+        fi
+        echo cache-files=$cache_files >> $GITHUB_ENV
+        echo requirements-file=$requirements_file >> $GITHUB_ENV
+      shell: bash
```

Comment:
I vote in favor of just always hashing both files. For a dev-dependency-only update, we will have to rebuild the pip packages unnecessarily, but I think that happens rarely enough.